src/io/inputs.jl

```diff
@@ -121,76 +122,75 @@ end
-    # instead, match by number of sites
-    dim_labels = Union{UnitRange{Int64},Vector{String}}[1:n for n in size(data)]
+    return loaded_data
+end
+
+"""
+Return vector of labels for each dimension.
+
+**Important:** Cannot trust indicated dimension metadata to get site labels,
+because this can be incorrect. Instead, match by number of sites.
+"""
+function _nc_dim_labels(
+    data_fn::String, data::Array{<:Real}, nc_file::NetCDF.NcFile
+)::Vector{Union{UnitRange{Int64},Vector{String}}}
+    local sites_idx::Int64
+
+    sites = "reef_siteid" in keys(nc_file.vars) ? _site_labels(nc_file) : 1:size(data, 2)
 
     try
-        # Obviously this will be an issue if any two dimensions have the same number of
-        # elements as the number of sites, but so far it hasn't happened...
-        i = first(findall(size(data) .== length(sites)))
+        # This will be an issue if two or more dimensions have the same number of elements
+        # as the number of sites, but so far it hasn't happened...
+        sites_idx = first(findall(size(data) .== length(sites)))
     catch err
         error(
-            "Error loading $data_fn : could not determine number of locations. Detected size: $(size(data)) | Known # sites: $(length(sites))",
+            "Error loading $data_fn : could not determine number of locations." *
+            "Detected size: $(size(data)) | Known number of locations: $(length(sites))",
         )
     end
-    dim_labels[i] = sites
 
-    try
-        loaded = NamedDimsArray(data; zip(dim_names, dim_labels)...)
-    catch err
-        if isa(err, KeyError)
-            n_sites = size(data, 2)
-            @warn "Provided file $(data_fn) did not have the expected dimensions (one of: timesteps, reef_siteid, scenarios)."
-            if n_sites != nrow(site_data)
-                error(
-                    "Mismatch in number of sites ($(data_fn)). Expected $(nrow(site_data)), got $(n_sites)",
-                )
-            end
-        else
-            rethrow(err)
-        end
-    end
+    dim_labels = Union{UnitRange{Int64},Vector{String}}[1:n for n in size(data)]
+    dim_labels[sites_idx] = sites
 
-    return loaded
+    return dim_labels
 end
 
 """
-    _char_to_string(vals)::Vector{String}
-
-Convert character array entries in netCDFs to string.
-
-Some packages used to write out netCDFs do not yet support
-string values, and instead reverts to writing out character arrays.
+Some packages used to write out netCDFs do not yet support string values, and instead
+reverts to writing out character arrays.
 """
-function _char_to_string(vals::AbstractVecOrMat)::Vector{String}
-    if vals isa Matrix
-        vals = map(x -> join(skipmissing(x)), eachcol(vals))
-    end
-
-    # R's ncdf4 package does not yet support string values
-    # so strip the null terminator from the joined character array.
-    vals = replace.(vals, "\0" => "")
-
-    return vals
+function _site_labels(nc_file::NetCDF.NcFile)::Vector{String}
```

Comment:
Location not sites ?

```suggestion
function _loc_labels(nc_file::NetCDF.NcFile)::Vector{String}
```

Reply:
In my opinion we should wait to change to location when we change in other places too. Right now the NamedDimsArray dimension is called `sites` so it makes sense for this one to be `sites_labels` as well and only change to `location_labels` or `loc_labels` when we also change the dimension name.